hackathon_site/dashboard/frontend/src/slices/event/teamSlice.ts

```diff
@@ -52,6 +58,84 @@ export const getCurrentTeam = createAsyncThunk<
     }
 });
 
+export const leaveTeam = createAsyncThunk<
+    Team,
+    void,
+    { state: RootState; rejectValue: RejectValue; dispatch: AppDispatch }
+>(
+    `${teamReducerName}/leaveTeam`,
+    async (_, { dispatch, getState, rejectWithValue }) => {
+        const teamCode = teamCodeSelector(getState());
+
+        try {
+            const response = await post<Team>("/api/event/teams/leave_team/", {
+                team_code: teamCode,
+            });
+
+            dispatch(
+                displaySnackbar({
+                    message: `You have left the team.`,
+                    options: { variant: "success" },
+                })
+            );
+
```

Comment:
here you can also dispatch the `closeTeamModalItem` action you created previously to close the side drawer

Reply:
same for the joinTeam thunk

Reply:
done